homeassistant/components/sensor/pjm.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for PJM data.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.pjm/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['beautifulsoup4==4.6.0']
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'http://datasnapshot.pjm.com/content/InstantaneousLoad.aspx'
+
+PJM_RTO_TOTAL = "PJM RTO Total"
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+ICON_POWER = 'mdi:flash'
+
+CONF_MONITORED_FEEDS = 'monitored_feeds'
+CONF_SENSOR_TYPE = 'type'
+CONF_INSTANTANEOUS_ZONE_LOAD = 'instantaneous_zone_load'
+CONF_INSTANTANEOUS_TOTAL_LOAD = 'instantaneous_total_load'
+CONF_ZONE = 'zone'
+
+SENSOR_TYPES = {
+    CONF_INSTANTANEOUS_ZONE_LOAD: ["PJM Instantaneous Zone Load", 'MW'],
+    CONF_INSTANTANEOUS_TOTAL_LOAD: ["PJM Instantaneous Total Load", 'MW'],
+}
+
+TYPES_SCHEMA = vol.In(SENSOR_TYPES)
+
+SENSORS_SCHEMA = vol.Schema({
+    vol.Required(CONF_SENSOR_TYPE): TYPES_SCHEMA,
+    vol.Optional(CONF_ZONE): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_FEEDS): [SENSORS_SCHEMA],
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the PJM sensor."""
+    websession = async_get_clientsession(hass)
+    dev = []
+
+    for variable in config[CONF_MONITORED_FEEDS]:
+        dev.append(PJMSensor(
+            hass.loop, websession, variable[CONF_SENSOR_TYPE],
+            variable.get(CONF_ZONE), variable.get(CONF_NAME)))
+
+    async_add_devices(dev, True)
+
+
+class PJMSensor(Entity):
+    """Implementation of a PJM sensor."""
+
+    def __init__(self, loop, websession, sensor_type, zone, name):
+        """Initialize the sensor."""
+        self.loop = loop
```

Comment:
Let's see what others say, but in general, we don't like Home Assistant to directly interact with data sources; this should be extracted to a package hosted on PyPi.